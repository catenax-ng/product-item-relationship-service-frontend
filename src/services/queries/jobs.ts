import { useMutation } from "@tanstack/react-query";
import { useQuery } from "../../lib/react-query";
import { IRSRequestBody } from "../../types/jobs";
import { getCurrentEnvironment } from "../../utils/sessionStorageHandling";
import { selectAPI } from "../api/jobs.api";

export const useFetchJobById = ({ id }: { id: string }, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, id], () => jobAPI.fetchJobById(id), {
    refetchInterval,
  });
};

export const useFetchJobs = (page: number, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, page], () => jobAPI.fetchJobs(page), { refetchInterval });
};

export const useCancelJobs = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation(jobAPI.cancelJob);
};

export const useCreateJob = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation({
    mutationFn: (data: IRSRequestBody) => {
      return jobAPI.createJob(data);
    },
  });
};
