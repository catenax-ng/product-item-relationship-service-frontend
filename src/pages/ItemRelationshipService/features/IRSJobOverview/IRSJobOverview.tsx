import { useState } from "react";
import { useFetchJobs } from "../../../../services/queries/jobs";
import { AutoRefreshSwitch } from "./components/AutoRefreshSwitch";
import { IRSJobTable } from "./components/IRSJobTable";
import { PaddedSection } from "../../../../components/layout/PaddedSection";

export const IRSJobOverview: React.FC = () => {
  const [isAutoRefreshEnabled, setIsAutoRefreshEnabled] = useState(false);
  const { isLoading, isError, data: jobs = [] } = useFetchJobs(isAutoRefreshEnabled ? 5000 : false);

  if (isError) {
    // TODO: Implement error handling
    return null;
  }

  return (
    <PaddedSection>
      <AutoRefreshSwitch onChange={setIsAutoRefreshEnabled} />
      <IRSJobTable isLoading={isLoading} jobs={jobs} />
    </PaddedSection>
  );
};
