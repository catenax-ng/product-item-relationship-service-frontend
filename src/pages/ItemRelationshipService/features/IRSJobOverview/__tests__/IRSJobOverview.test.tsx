import { JobStatusResultSuccess } from "../../../../../services/api/__test__/jobStatusResult.example";
import * as queryHooks from "../../../../../services/queries/jobs";
import { render } from "../../../../../testing/test-utils";
import { IRSJobOverview } from "../IRSJobOverview";

it("renders a loading state", () => {
  const mockData = {
    data: undefined,
    isError: false,
    isLoading: true,
  };
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  //@ts-ignore
  jest.spyOn(queryHooks, "useFetchJobs").mockReturnValue(mockData);
  const { container } = render(<IRSJobOverview />);
  expect(container).toMatchSnapshot();
});

it("renders a error state", () => {
  const mockData = {
    data: undefined,
    isError: true,
    isLoading: false,
  };
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  //@ts-ignore
  jest.spyOn(queryHooks, "useFetchJobs").mockReturnValue(mockData);
  const { container } = render(<IRSJobOverview />);
  expect(container).toMatchSnapshot();
});

it("renders a data state", () => {
  const mockData = {
    data: JobStatusResultSuccess,
    isError: false,
    isLoading: false,
  };
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  //@ts-ignore
  jest.spyOn(queryHooks, "useFetchJobs").mockReturnValue(mockData);
  const { container } = render(<IRSJobOverview />);
  expect(container).toMatchSnapshot();
});
