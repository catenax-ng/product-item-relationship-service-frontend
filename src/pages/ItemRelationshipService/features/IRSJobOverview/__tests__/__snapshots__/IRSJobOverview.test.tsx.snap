// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a data state 1`] = `
<div>
  <section
    class="css-jlkg07"
  >
    <span
      class="MuiSwitch-root MuiSwitch-sizeMedium css-113qkos-MuiSwitch-root"
    >
      <span
        class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
        data-testid="IOSSwitch"
      >
        <input
          class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
          type="checkbox"
        />
        <span
          class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
      </span>
      <span
        class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
    content.irs.jobsTable.toggleAutoRefresh
    <div
      class="MuiBox-root css-b7ealj"
    >
      <div
        aria-colcount="5"
        aria-multiselectable="false"
        aria-rowcount="5"
        class="irs-table MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard css-13itlwa-MuiDataGrid-root"
        role="grid"
      >
        <div>
          <div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-xgx8td"
              >
                <div
                  class="MuiBox-root css-70qvj9"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                  >
                    content.irs.jobsTable.title
                    <span
                      class="MuiBox-root css-dah8a6"
                    >
                      (
                      0
                      )
                    </span>
                  </h5>
                </div>
                <div
                  class="MuiBox-root css-1bvc4cc"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
        >
          <div
            class="MuiDataGrid-columnHeaders css-f3jnds-MuiDataGrid-columnHeaders"
            style="min-height: 57px; max-height: 57px; line-height: 57px;"
          >
            <div
              aria-rowindex="1"
              class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-1s0hp0k-MuiDataGrid-columnHeadersInner"
              role="row"
              style="transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-colindex="1"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                data-field="jobId"
                role="columnheader"
                style="width: 0px; min-width: 0; max-width: 0;"
                tabindex="0"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        content.irs.jobsTable.jobID
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 57px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="2"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                data-field="startedOn"
                role="columnheader"
                style="width: 150px; min-width: 150px; max-width: 150px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        content.irs.jobsTable.startDate
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 57px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="3"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                data-field="jobCompleted"
                role="columnheader"
                style="width: 150px; min-width: 150px; max-width: 150px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        content.irs.jobsTable.endDate
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 57px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            style="overflow: visible; width: 0px;"
          >
            <div
              class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
              style="width: 0px; height: auto; margin-top: 57px; overflow-x: hidden; overflow-y: hidden;"
            >
              <div
                class="MuiDataGrid-virtualScrollerContent css-1kwdphh-MuiDataGrid-virtualScrollerContent"
                style="width: auto; height: 228px; min-height: auto;"
              >
                <div
                  class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                  style="transform: translate3d(0px, 0px, 0px);"
                >
                  <div
                    aria-rowindex="2"
                    aria-selected="false"
                    class="MuiDataGrid-row"
                    data-id="demo-initial"
                    data-rowindex="0"
                    role="row"
                    style="max-height: 57px; min-height: 57px;"
                  >
                    <div
                      aria-colindex="1"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="0"
                      data-field="jobId"
                      role="cell"
                      style="min-width: 0; max-width: 0; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        demo-initial
                      </div>
                    </div>
                    <div
                      aria-colindex="2"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="1"
                      data-field="startedOn"
                      role="cell"
                      style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        2022-11-14 09:46:51
                      </div>
                    </div>
                    <div
                      aria-colindex="3"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="2"
                      data-field="jobCompleted"
                      role="cell"
                      style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        2022-11-14 09:50:16
                      </div>
                    </div>
                  </div>
                  <div
                    aria-rowindex="3"
                    aria-selected="false"
                    class="MuiDataGrid-row"
                    data-id="demo-running"
                    data-rowindex="1"
                    role="row"
                    style="max-height: 57px; min-height: 57px;"
                  >
                    <div
                      aria-colindex="1"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="0"
                      data-field="jobId"
                      role="cell"
                      style="min-width: 0; max-width: 0; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        demo-running
                      </div>
                    </div>
                    <div
                      aria-colindex="2"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="1"
                      data-field="startedOn"
                      role="cell"
                      style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        2022-11-14 09:46:51
                      </div>
                    </div>
                    <div
                      aria-colindex="3"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="2"
                      data-field="jobCompleted"
                      role="cell"
                      style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        2022-11-14 09:50:16
                      </div>
                    </div>
                  </div>
                  <div
                    aria-rowindex="4"
                    aria-selected="false"
                    class="MuiDataGrid-row"
                    data-id="demo-completed"
                    data-rowindex="2"
                    role="row"
                    style="max-height: 57px; min-height: 57px;"
                  >
                    <div
                      aria-colindex="1"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="0"
                      data-field="jobId"
                      role="cell"
                      style="min-width: 0; max-width: 0; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        demo-completed
                      </div>
                    </div>
                    <div
                      aria-colindex="2"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="1"
                      data-field="startedOn"
                      role="cell"
                      style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        2022-11-14 09:46:51
                      </div>
                    </div>
                    <div
                      aria-colindex="3"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="2"
                      data-field="jobCompleted"
                      role="cell"
                      style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        2022-11-14 09:50:16
                      </div>
                    </div>
                  </div>
                  <div
                    aria-rowindex="5"
                    aria-selected="false"
                    class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                    data-id="demo-canceled"
                    data-rowindex="3"
                    role="row"
                    style="max-height: 57px; min-height: 57px;"
                  >
                    <div
                      aria-colindex="1"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="0"
                      data-field="jobId"
                      role="cell"
                      style="min-width: 0; max-width: 0; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        demo-canceled
                      </div>
                    </div>
                    <div
                      aria-colindex="2"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="1"
                      data-field="startedOn"
                      role="cell"
                      style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        2022-11-14 09:46:51
                      </div>
                    </div>
                    <div
                      aria-colindex="3"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="2"
                      data-field="jobCompleted"
                      role="cell"
                      style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        2022-11-14 09:50:16
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Mui-resizeTriggers"
          >
            <div
              class="expand-trigger"
            >
              <div
                style="width: 1px; height: 1px;"
              />
            </div>
            <div
              class="contract-trigger"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`renders a error state 1`] = `
<div>
  <section>
    <div
      class="irs-error-page MuiBox-root css-0"
    >
      <div
        class="irs-error-page-header MuiBox-root css-0"
      >
        <h1>
          global.errors.title
        </h1>
      </div>
      <div
        class="error-page-body MuiBox-root css-0"
      >
        <p>
          global.errors.description
        </p>
        <p>
          <i>
            test Error
          </i>
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`renders a loading state 1`] = `
<div>
  <section
    class="css-jlkg07"
  >
    <span
      class="MuiSwitch-root MuiSwitch-sizeMedium css-113qkos-MuiSwitch-root"
    >
      <span
        class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
        data-testid="IOSSwitch"
      >
        <input
          class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
          type="checkbox"
        />
        <span
          class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
      </span>
      <span
        class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
    content.irs.jobsTable.toggleAutoRefresh
    <div
      class="MuiBox-root css-b7ealj"
    >
      <div
        aria-colcount="5"
        aria-multiselectable="false"
        aria-rowcount="1"
        class="irs-table MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard css-13itlwa-MuiDataGrid-root"
        role="grid"
      >
        <div>
          <div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-xgx8td"
              >
                <div
                  class="MuiBox-root css-70qvj9"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                  >
                    content.irs.jobsTable.title
                    <span
                      class="MuiBox-root css-dah8a6"
                    >
                      (
                      0
                      )
                    </span>
                  </h5>
                </div>
                <div
                  class="MuiBox-root css-1bvc4cc"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
        >
          <div
            style="height: auto; width: 0px; position: absolute; top: 57px; bottom: 0px; z-index: 4; pointer-events: none;"
          >
            <div
              class="MuiDataGrid-overlay css-1lk5ll6-MuiDataGrid-overlay"
            >
              <span
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
                role="progressbar"
                style="width: 40px; height: 40px;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="MuiDataGrid-columnHeaders css-f3jnds-MuiDataGrid-columnHeaders"
            style="min-height: 57px; max-height: 57px; line-height: 57px;"
          >
            <div
              aria-rowindex="1"
              class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-1s0hp0k-MuiDataGrid-columnHeadersInner"
              role="row"
              style="transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-colindex="1"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                data-field="jobId"
                role="columnheader"
                style="width: 0px; min-width: 0; max-width: 0;"
                tabindex="0"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        content.irs.jobsTable.jobID
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 57px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="2"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                data-field="startedOn"
                role="columnheader"
                style="width: 150px; min-width: 150px; max-width: 150px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        content.irs.jobsTable.startDate
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 57px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="3"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                data-field="jobCompleted"
                role="columnheader"
                style="width: 150px; min-width: 150px; max-width: 150px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        content.irs.jobsTable.endDate
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 57px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            style="overflow: visible; width: 0px;"
          >
            <div
              class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
              style="width: 0px; height: auto; margin-top: 57px; overflow-x: hidden; overflow-y: hidden;"
            >
              <div
                class="MuiDataGrid-virtualScrollerContent css-1kwdphh-MuiDataGrid-virtualScrollerContent"
                style="width: auto; height: 114px; min-height: auto;"
              >
                <div
                  class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                  style="transform: translate3d(0px, undefinedpx, 0px);"
                />
              </div>
            </div>
          </div>
          <div
            class="Mui-resizeTriggers"
          >
            <div
              class="expand-trigger"
            >
              <div
                style="width: 1px; height: 1px;"
              />
            </div>
            <div
              class="contract-trigger"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
